presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/SqlInvokedScalarFromAnnotationsParser.java

```diff
@@ -101,7 +101,7 @@ private SqlInvokedScalarFromAnnotationsParser() {}
     {
         Set<Method> methods = findPublicStaticMethods(
                 clazz,
-                ImmutableSet.of(SqlInvokedScalarFunction.class, SqlType.class, SqlParameter.class, SqlParameters.class, Description.class),
+                ImmutableSet.of(SqlInvokedScalarFunction.class),
```

Comment:
This is not a bug. It is intended. For each method, we are looking for methods that are annotated with any of those functions-related annotation. and not just @SqlInvokedScalarFunction. For example, if a function is annotated with @SqlType @Description, but does not have @SqlInvokedScalarFunction, we should throw, instead of silently ignore the method.

This exactly same logic can be found in `ScalarFromAnnotationsParser` for @ScalarFunction, and we should preserve the behavior.

The annotation parsers were written in a way such that it assume @ScalarFunction and @SqlInvokedScalarFunction are two disjoint things, and a deviation from that assumption will require more code than simply the change of this line.

Reply:
Why? Isn't `@SqlInvokedScalarFunction` required to define an sql invoked scalar function? If yes, why do we need to assume otherwise? I mean it's nice to catch incorrectly implemented functions. But assuming the functions have test cases, I'm sure without the check here it can also be caught. The reason this didn't work is because `Description` and `SqlType` can also be used to annotate `ScalarFunction`. I guess removing these 2 will make it work as well if you prefer. I don't see any valid reason to look for all of them.

Reply:
I don't see why we have to enforce a class either only have Java implementation or only have SQL implementation. I think it's better to have all functions with the same name to be in the same class.

Reply:
Anyways, I changed the registration to use `scalars` instead of `scalar` for the java implementation so now it will work without changing the parser. It's totally ok to have java and sql functions mixed as long as all annotations are on methods.

Reply:
Sure, I'm fine as long as this intended check is restored.